internal/runners/initialize/init.go

```diff
@@ -33,19 +32,49 @@ type RunParams struct {
 
 // Initialize stores scope-related dependencies.
 type Initialize struct {
-	out output.Outputer
+	config projectfile.ConfigGetter
+	out    output.Outputer
 }
 
 type primeable interface {
+	primer.Configurer
 	primer.Outputer
 }
 
 // New returns a prepared ptr to Initialize instance.
 func New(prime primeable) *Initialize {
-	return &Initialize{prime.Output()}
+	return &Initialize{prime.Config(), prime.Output()}
 }
 
-func sanitize(params *RunParams) error {
+// inferLanguage tries to infer a reasonable default language from the project currently in use
+// (i.e. `state use show`).
+// Error handling is not necessary because it's an input error to not include a language to
+// `state init`. We're just trying to infer one as a convenience to the user.
+func inferLanguage(config projectfile.ConfigGetter) string {
+	defaultProjectDir := config.GetString(constants.GlobalDefaultPrefname)
+	if defaultProjectDir == "" {
+		return ""
+	}
+	defaultProj, err := project.FromPath(defaultProjectDir)
+	if err != nil {
+		return ""
+	}
+	commitID := defaultProj.CommitUUID()
+	if commitID == "" {
+		return ""
+	}
+	lang, err := model.FetchLanguageForCommit(commitID)
+	if err != nil {
+		return ""
+	}
+	return fmt.Sprintf("%s@%s", lang.Name, lang.Version)
```

Comment:
It feels a bit awkward that we're interpolating this just to then parse it again a bit later. Feels like we could just return the language struct value here and use it accordingly in the consuming code.

Reply:
Again, we're "sanitizing" parameters for use in a later function. I agree it's a bit awkward at first glance, but a more involved refactor may be out of scope here.

Reply:
Yeah that's a very fair point, but seeing as the story is about stabilizing I think it's fair game.